File: android/app/src/main/java/com/demo/videopick/data/model/SavedLocation.kt
package com.demo.videopick.data.model

import kotlinx.serialization.Serializable
import java.util.UUID

@Serializable
data class SavedLocation(
    val id: String = UUID.randomUUID().toString(),
    val name: String,
    val latitude: Double,
    val longitude: Double,
    val createdAt: Long = System.currentTimeMillis(),
) {
    val coordinateText: String
        get() = "%.6f,%.6f".format(latitude, longitude)
}

File: ios/DouyinDownLoad/Models/SavedLocation.swift
import Foundation

struct SavedLocation: Codable, Identifiable, Hashable {
    var id: UUID
    var name: String
    var latitude: Double
    var longitude: Double
    var createdAt: Date

    init(id: UUID = UUID(), name: String, latitude: Double, longitude: Double, createdAt: Date = Date()) {
        self.id = id
        self.name = name
        self.latitude = latitude
        self.longitude = longitude
        self.createdAt = createdAt
    }

    var coordinateText: String {
        String(format: "%.6f,%.6f", latitude, longitude)
    }
}
